### ISSUE TITLE
Breaking out matrix A with free/constrained sections

### ISSUE THREAD

**Issue by oveddan** [danielepanozzo/gp-old#14]
In the section4 recitation slide, it shows breaking out matrix A into sections that correspond to free/constrained variables:

<img width="241" alt="screen shot 2017-04-10 at 6 19 15 pm" src="https://cloud.githubusercontent.com/assets/891755/24888782/8c9aac7e-1e1a-11e7-8aa6-2ddf9ccfd5fd.png">

What parts of matrix A correspond to `Acf` and `Afc`?

**Comment by jpanetta** [comment]
The slides were written assuming the variables were ordered so that all free variables appear first followed by all constrained variables. Then A can be written as the block matrix indicated above where A_{ff} are the first numFree rows and columns, A_cc are the last numFixed rows and columns, A_{fc} are the first numFree rows of the last numFixed columns, and A_{cf} are the last numFixed rows of the first numFree columns.

But this ordering was only used to simplify the derivation. For arbitrary ordering, you can compute A_{ff} by removing from A the rows and cols corresponding to constrained variables. You can get A_{fc} by copying from A the columns corresponding to the constrained variables and then removing from them the rows corresponding to the constrained variables.

**Comment by oveddan** [comment]
Thank you, this makes sense.

**Comment by oveddan** [comment]
What exactly is a row corresponding to a constrained variable?  Is it any row that has a value for the constrained variable? In our case a row for a face that corresponds to the constrained vertex?

**Comment by oveddan** [comment]
Or maybe, since our K is `G^T * G` - any row in the resulting matrix that corresponds to the vertex?

**Comment by jpanetta** [comment]
Yes, for the case of matrix G^T G, row i corresponds to vertex i and column
j to vertex j. The ith row of G^T G u is the value computed at vertex i of
some type of laplacian of per-vertex scalar field u.

But you can also apply this technique to the per-triangle-vector smoothing
problem (with complex matrix Q).

The general statement for real vars is that for any quadratic form:
x^T A x = sum_ij x_i A_ij x_j
row i of A is paired with variable i and col j is paired with variable j.
This pairing is what I meant when I said "corresponding," and the
derivation on slide 30 should make the pairing's significance clear. (For
example, entry A_ij appears only in constant terms when x_i and x_j are
constrained, so it can be dropped).

For complex vars, the same reasoning holds, but you must replace the
transpose with the conjugate-transpose.

-Julian


On Tue, Apr 11, 2017 at 11:39 AM Dan Oved <notifications@github.com> wrote:

Or maybe, since our K is a dot product G^T * G - any row in the resulting
matrix that corresponds to the vertex?

—
You are receiving this because you commented.


Reply to this email directly, view it on GitHub
<https://github.com/danielepanozzo/gp/issues/14#issuecomment-293303757>,
or mute
the thread
